packages/remix-dev/vite/plugin.ts

```diff
@@ -218,6 +211,13 @@ type BuildContext =
       serverBundleId: string | undefined;
```

Comment:
Is there a reason this is `string | undefined` instead of `?: string`?

Reply:
So that you're forced to add the key to the object, even if the value is undefined.